python/lsst/sims/catalogs/generation/db/dbConnection.py

```diff
@@ -542,24 +542,31 @@ def _postprocess_results(self, results):
             * _final_pass(retresults) : the result of calling the _final_pass method on a
               structured array constructed from the query data.
         """
+
         if len(results) > 0:
             cols = [str(k) for k in results[0].keys()]
         else:
             return results
+
         dtype = numpy.dtype([(k,)+self.typeMap[k] for k in cols])
+
         if len(set(cols)&set(self.dbDefaultValues)) > 0:
-            retresults = numpy.empty((len(results),), dtype=dtype)
-            for i, result in enumerate(results):
-                for k in cols:
-                    if k in self.dbDefaultValues and not result[k]:
-                        retresults[i][k] = self.dbDefaultValues[k]
-                    else:
-                        try:
-                            retresults[i][k] = result[k]
-                        except TypeError:
-                            raise TypeError("Couldn't convert column %s"%(k))
+
+            for colName in cols:
+                if colName not in results[0].keys() and colName not in self.dbDefaultValues:
```

Comment:
Yes.  I'll be honest: I am not actually sure how that TypeError could ever have been raised, even before this pull request.

I assumed it was there for a reason.

Maybe it was there in case there was a type mismatch between results and retresults.

Regardless, I will drop this check (lines 555 through 557).